docs(select): update select docs (#2031)
IgniteUI/igniteui-docfx: en/components/select.md

---

## Select

> <p class="highlight">The [IgxSelectComponent]({environment:angularApiUrl}/classes/igxselectcomponent.html) allows a single selection from a list of items, placed in a dropdown. This Material-based form control offers quick items list navigation, including selection, based on a single or multiple characters match. Since the `igx-select` extends the`igx-drop-down`, you can handle all of the dropdown events.</p>

Reviewer: ```suggestion
<p class="highlight">The [Select Component]({environment:angularApiUrl}/classes/igxselectcomponent.html) allows a single selection from a list of items, placed in a dropdown. This Material-based form control offers quick items list navigation, including selection, based on a single or multiple characters match. Since `IgxSelectComponent` extends `IgxDropDownComponent`, you can handle all of the Drop Down events in the Select instance as well.</p>
```

> <p class="highlight">The [SelectComponent]({environment:angularApiUrl}/classes/igxselectcomponent.html) allows a single selection from a list of items, placed in a dropdown. This Material-based form control offers quick items list navigation, including selection, based on a single or multiple characters match. Since the [IgxSelectComponent]({environment:angularApiUrl}/classes/igxselectcomponent.html) extends the [IgxDropDownComponent]({environment:angularApiUrl}/classes/igxdropdowncomponent.html), you can handle all of the Drop Down events in the Select instance as well.</p>

Reviewer: Replace `SelectComponent` with `Select Component` in all instances.

### Demo
<div class="sample-container loading" style="height: 300px;">
    <iframe id="select-sample-1-iframe" frameborder="0" seamless width="100%" height="100%" src="{environment:demosBaseUrl}/data-entries/select-sample-1" onload="onSampleIframeContentLoaded(this);"></iframe>
</div>

<button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="select-sample-1-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>

### Usage

> To get started with the [IgxSelectComponent]({environment:angularApiUrl}/classes/igxselectcomponent.html), first you need to import the **IgxSelectModule** in your **app.module**:

Reviewer: ```suggestion
To get started with the [Select Component]({environment:angularApiUrl}/classes/igxselectcomponent.html), first you need to import the **IgxSelectModule** in your **app.module**:
```

export class AppModule {}
```

> Then add the `igx-select` along with a list of items to choose from. We use an [`IgxSelectItemComponent`]({environment:angularApiUrl}/classes/igxselectitemcomponent.html) to display the items that the `igx-select` contains.

Reviewer: ```suggestion
Then add `igx-select` along with a list of items to choose from. We use [`igx-select-item`]({environment:angularApiUrl}/classes/igxselectitemcomponent.html) to display the items that `igx-select` contains.
```

</igx-select>
```

> Another way to do it would be to use a collection of the items that we want to display, using the [ngForOf](https://angular.io/api/common/NgForOf) directive:

Reviewer: ```suggestion
Another way to do it would be to use a collection of items that we want to display using the [*ngFor](https://angular.io/api/common/NgForOf) structural directive:
```

        <igx-select-item *ngFor="let item of group.items" [value]="item.type" [text]="item.type">
            {{item.type}}
            <igx-icon 
                title="Local product"
                class="icon"
                color="green"
                *ngIf="item.origin === 'local';else templateImport"
            >local_shipping</igx-icon>
            <ng-template #templateImport>
                <igx-icon
                    title="Import product"
                    class="icon"
                    color="orange"
                >flight</igx-icon>
            </ng-template>
        </igx-select-item>
    </igx-select-item-group>
</igx-select>
```

<div class="sample-container loading" style="height: 400px;">
    <iframe id="select-sample-2-iframe" frameborder="0" seamless="" width="100%" height="100%" data-src="{environment:demosBaseUrl}/data-entries/select-sample-2" class="lazyload"></iframe>
</div>

<button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="select-sample-2-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>

#### Header & Footer

> Currently, there are no default header and footer templates for the `igxSelect` component. However, you can add a header or a footer template by marking them respectively with `IgxSelectHeaderDirective` or `IgxSelectFooterDirective`. As these are custom templates, you should define the styling as well.

Reviewer: ```suggestion
Currently, there are no default header and footer templates for the Select component. However, you can add a header or a footer template by marking them respectively with `IgxSelectHeaderDirective` or `IgxSelectFooterDirective`. As these are custom templates, you should define the styling as well.
```

                >flight</igx-icon>
            </ng-template>
        </igx-select-item>
    </igx-select-item-group>
</igx-select>
```

<div class="sample-container loading" style="height: 400px;">
    <iframe id="select-sample-2-iframe" frameborder="0" seamless="" width="100%" height="100%" data-src="{environment:demosBaseUrl}/data-entries/select-sample-2" class="lazyload"></iframe>
</div>

<button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="select-sample-2-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>

#### Header & Footer
Currently, there are no default header and footer templates for the Select. However, you can add a header or a footer template by marking them respectively with `IgxSelectHeader` or `IgxSelectFooter`. As these are custom templates, you should define their styling as well.

In this example, there are both header and footer ng-templates defined. In the header there is a basic filtering, implemented via [`igx-buttongroup`]({environment:angularApiUrl}/classes/igxbuttongroupcomponent.html). The footer includes static summary of all of the items, based on the delivery method.

```html

>     <igx-select>

Reviewer: This is still indented when it shouldn't be.

                <span class="sample-template-heading">TOTAL</span>
                <div class="sample-template-icons">
                    <span class="sample-template-icons__item">
                        <igx-icon

>                         title="flight"

Reviewer 2: indentation again

<button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="select-header-footer-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>

#### Custom Toggle Button

> You can customize the default toggle button, using the `IgxSelectToggleIcon` directive or setting a `TemplateRef` to the [`toggleIconTemplate`]({environment:angularApiUrl}/classes/igxselectcomponent.html#toggleicontemplate) property.

Reviewer: ```suggestion
You can customize the default toggle button, using the `igxSelectToggleIcon` directive or setting a `TemplateRef` to the [`toggleIconTemplate`]({environment:angularApiUrl}/classes/igxselectcomponent.html#toggleicontemplate) property.
```

```html
<igx-select #select>
    ...
    <ng-template igxSelectToggleIcon let-collapsed>
        <igx-icon>{{ collapsed ? 'add_circle' : 'add_circle_outline'}}</igx-icon>
    </ng-template>
    ...
<igx-select>
```

### Keyboard Navigation

- Open the `igx-select` by clicking on the `Space`, `Enter` or `ALT + Up/Down Arrow` keys, while the select is focused.
- Close the `igx-select` using the `ALT + Up/Down Arrow` combination or any of the `Enter`, `Space`, `Esc` or `Tab` keys.
- Using the `Up/Down Arrow` keys will navigate through the items.
- Using the `Home` or `End` keys will navigate you to the first and last items in the list.
- You can navigate through list items, starting with a certain character, by pressing the corresponding key.
- You can navigate to a specific item by rapidly typing the first few characters of the item you wish to go to.
- Select an item using the `Enter` or `Space` keys

>[!NOTE]

> >`igx-select` supports only a *single* selection of items.

Reviewer: ```suggestion
>`igx-select` supports only *single* selection of items.
```

    /* -- */
}
```
>[!NOTE]

> >If you pass in your custom settings both as an argument in the `open` function and into the template, the `igx-select` will use the ones, provided in the `open` function. However, if you bind the settings to an internal event, such as `onOpening` or `onOpened` then `igx-select` will use the settings that are provided in the template.

Reviewer: ```suggestion
>If you pass in your custom settings both as an argument in the `open` function and in the template, `igx-select` will use the ones provided in the `open` function. However, if you bind the settings to an internal event, such as `onOpening` or `onOpened` then `igx-select` will use the settings that are provided in the template.
```

### Styling
Every component has its own theme.

To get the `IgxSelect` styled, you have style its containing components. In our case, we need to use both the [igx-input-group-theme]({environment:sassApiUrl}/index.html#function-igx-input-group-theme) and the [igx-drop-down-theme]({environment:sassApiUrl}/index.html#function-igx-drop-down-theme).

> Take a look at the [`igxInputGroup`](input_group.md#styling) and the [`igxDropdown`](drop_down.md#styling) styling sections to get a better understanding of how to style those two components.

Reviewer: ```suggestion
Take a look at the [`Input Group`](input_group.md#styling) and [`Drop Down`](drop_down.md#styling) styling sections to get a better understanding of how to style those two components.
```
